Sources/Client/Client/Client+Events.swift

```diff
@@ -49,16 +49,21 @@ public final class SubscriptionBag: Cancellable {
     
     /// Add a subscription.
     /// - Parameter subscription: a subscriiption.
-    public func add(_ subscription: Cancellable) {
-        subscriptions.append(subscription)
+    public func add(_ subscription: Cancellable?) {
```

Comment:
Why do need to have these parameters optional? (same in `adding`)

Reply:
It's more flexible for usage in chain requests:
```swift
subscriptionBag.add(request { [weak self] in
    self?.subscriptionBag.add(self?.subRequest { ... })
})
```

Reply:
I think we should use the weak self dance in these situations:
```swift
guard let self = self else { return }
```
since it's not natural to have `add(Cancellable?)` into `SubscriptionBag`